attestation-service/src/main/kotlin/AttestationService.kt

```diff
@@ -105,14 +201,66 @@ data class IOSAttestationConfiguration @JvmOverloads constructor(
             /**
              * @see AppData.iosVersionOverride
              */
-            fun overrideIosVersion(version: String) = apply { iosVersionOverride = version }
+            fun overrideIosVersion(version: OsVersions) = apply { iosVersionOverride = version }
 
             fun build() = AppData(teamIdentifier, bundleIdentifier, sandbox, iosVersionOverride)
         }
     }
 
 }
 
+typealias ParsedVersions = Pair<SemVer?, BuildNumber?>
+
+/**
+ * iOS build number. As per [TidBITS.com](https://tidbits.com/2020/07/08/how-to-decode-apple-version-and-build-numbers/):
+ *
+ * An Apple build number also has three parts:
+ *
+ * *  Major version: Within Apple, the major version is called the build train.
+ * *  Minor version: For iOS and its descendants, the minor version tracks with the minor release; for macOS, it tracks with patch releases.
+ * *  Daily build version: The daily build indicates how many times Apple has built the source code for the release since the previous public release.
+ *
+ * While this last bit about the daily build number is phrased somewhat fuzzy, it really is a strictly increasing decimal number.
+ */
+class BuildNumber private constructor(val buildTrain: UInt, val minorVersion: String, val buildVer: UInt) :
+    Comparable<BuildNumber> {
+
+
+    constructor(buildNumber: String) : this(parseBuildNumber(buildNumber))
+
+    constructor(boxed: Triple<UInt, String, UInt>) : this(boxed.first, boxed.second, boxed.third)
+
+
+    /**
+     * Integer representation of the build number. Converts [buildTrain] into a hex number, concatenates it with [minorVersion] radix-36-parsed
```

Comment:
interesting, it's realliy radix 36?

Reply:
0-9, A-Z always uppercase in that order looks like base36 to me ;-)